packages/task/src/node/process/process-task-runner.ts

```diff
@@ -76,17 +76,15 @@ export class ProcessTaskRunner implements TaskRunner {
 
         // sanity checks:
         // - we expect the cwd to be set by the client.
-        if (!taskConfig.cwd) {
+        if (!options || !options.cwd) {
             throw new Error("Can't run a task when 'cwd' is not provided by the client");
         }
 
-        const cwd = this.asFsPath(taskConfig.cwd);
         // Use task's cwd with spawned process and pass node env object to
         // new process, so e.g. we can re-use the system path
-        options = {
-            cwd: cwd,
-            env: process.env
-        };
+        if (options && !options.env) {
+            Object.assign(options, { env: process.env });
```

Comment:
```ts
if (options) {
    options.env = {
        ...process.env,
        ...options.env,
    };
}
```

Because I don't think we want to remove all the current environments variables and have the user specify all of them. More common use case is to add/override only a few. Is VS Code doing differently?

Reply:
Thank you for the comment. Resolved.